From ea142bf9919040dde4941b0e0dba23b0d28c61c2 Mon Sep 17 00:00:00 2001
From: rpm-build <rpm-build>
Date: Sat, 1 Jun 2024 19:26:14 -0300
Subject: [PATCH] winex11: always update display cache

Fix virtual desktop custom resolutions from explorer /desktop command line
---
 dlls/winex11.drv/window.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index 3d350f4..34e2638 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -2037,7 +2037,9 @@ void X11DRV_SetDesktopWindow( HWND hwnd )
 
     if (!width && !height)  /* not initialized yet */
     {
-        RECT rect = NtUserGetVirtualScreenRect();
+        RECT rect;
+        NtUserCallNoParam( NtUserCallNoParam_DisplayModeChanged );
+        rect = NtUserGetVirtualScreenRect();
 
         SERVER_START_REQ( set_window_pos )
         {
-- 
2.45.2

